vsphere/vsphere.go

```diff
@@ -732,3 +742,134 @@ func isExponentialError(err error) bool {
 	}
 	return false
 }
+
+type progressLogger struct {
+	prefix string
+
+	wg sync.WaitGroup
+
+	sink chan chan progress.Report
+	done chan struct{}
+}
+
+func newProgressLogger(prefix string) *progressLogger {
+	p := &progressLogger{
+		prefix: prefix,
+
+		sink: make(chan chan progress.Report),
+		done: make(chan struct{}),
+	}
+
+	p.wg.Add(1)
+
+	go p.loopA()
+
+	return p
+}
+
+func createProgressLogger(prefix string) *progressLogger {
+	return newProgressLogger(prefix)
+}
+
+type OutputFlag struct {
+	common
+
+	JSON bool
+	XML  bool
+	TTY  bool
+	Dump bool
+	Out  io.Writer
+
+	formatError  bool
+	formatIndent bool
+}
+
+// Type to help flags out with only registering/processing once.
+type common struct {
+	register sync.Once
+	process  sync.Once
+}
+
+func (c *common) RegisterOnce(fn func()) {
+	c.register.Do(fn)
+}
+
+func (c *common) ProcessOnce(fn func() error) (err error) {
+	c.process.Do(func() {
+		err = fn()
+	})
+	return err
+}
+
+// loopA runs before Sink() has been called.
+func (p *progressLogger) loopA() {
```

Comment:
why do we need two loops and a channel of channel ?
Can't we only use loopB and let Sink() return a channel p.sink where p.sink = make(chan progress.Report) ?

Reply:
removed the progress logger. It was optional and overly complex. Had inherited this from the govc sample code.